Some ARMv7 cleanups and add UEFI support for ARMv7

nullr0ute opened the pull request:
This PR drops a few old legacy code paths that haven't been needed for some time as well  as adding support for UEFI on ARMv7

ausil commented on blivet/formats/disklabel.py:
```diff
@@ -223,8 +223,12 @@ def get_platform_label_types(cls):
+            label_types = ["gpt", "msdos"]
         elif arch.is_efi() and not arch.is_aarch64():
             label_types = ["gpt"]
+        elif arch.is_efi() and arch.is_arm():
```
this section should be moved above, as  it has efi and the arch is not aarch64, we should never get this far

nullr0ute replied:
Good catch

vojtechtrefny commented:
Jenkins, test this please.

vojtechtrefny commented on blivet/arch.py:
```diff
@@ -182,27 +182,6 @@ def is_aarch64():
-    if not is_arm():
-        return None
-
-    if flags.arm_platform:
```
Please also remove the (now unused) flags import to make pylint happy.

nullr0ute replied:
Will do, update coming in shortly, I'll also remove the "self.arm_platform = None" declare from flags.py too

dwlehman commented on tests/formats_test/disklabel_test.py:
```diff
@@ -80,7 +80,7 @@ def test_platform_label_types(self, arch):
 
         arch.is_efi.return_value = True
-        self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt"])
+        self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt", "msdos"])
```
I think you need to use a mock for `arch.is_arm` to make this work, since on non-arm this test _should_ fail due to the return value being `["gpt"]`. I suggest testing `get_platform_label_types` for both possible values of `arch.is_arm`.

nullr0ute replied:
@dwlehman technically the UEFI standard supports either gpt or msdos no matter what the architecture

dwlehman replied:
That's not as important as the fact that `get_platform_label_types` currently returns ['gpt'] for non-arm, non-aarch64 UEFI platforms. The unit tests need to be updated to take into account `arch.is_arm` since you added logic around it to `get_platform_label_types`. It passes right now only because `disklabel.arch.is_arm` is a `Mock` which by default returns something that evaluates to True, meaning that `test_platform_label_types` will behave as if the test platform is arm all the time.

dwlehman replied:
Here's what I have in mind:

```diff
diff --git a/tests/formats_test/disklabel_test.py b/tests/formats_test/disklabel_test.py
index 21c2a748..4060e731 100644
--- a/tests/formats_test/disklabel_test.py
+++ b/tests/formats_test/disklabel_test.py
@@ -70,6 +70,7 @@ class DiskLabelTestCase(unittest.TestCase):
 
         arch.is_s390.return_value = False
         arch.is_efi.return_value = False
+        arch.is_arm.return_value = False
         arch.is_aarch64.return_value = False
         arch.is_pmac.return_value = False
 
@@ -80,9 +81,16 @@ class DiskLabelTestCase(unittest.TestCase):
         arch.is_pmac.return_value = False
 
         arch.is_efi.return_value = True
+        self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt"])
+        arch.is_arm.return_value = True
         self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt", "msdos"])
+        arch.is_arm.return_value = False
         arch.is_efi.return_value = False
 
+        arch.is_aarch64.return_value = True
+        self.assertEqual(disklabel_class.get_platform_label_types(), ["msdos", "gpt"])
+        arch.is_aarch64.return_value = False
+
         arch.is_s390.return_value = True
         self.assertEqual(disklabel_class.get_platform_label_types(), ["msdos", "dasd"])
         arch.is_s390.return_value = False
```

dwlehman commented:
Generally, will these changes break older hardware? Is that hardware still in reasonably wide use?

nullr0ute commented:
So I think everything is now correct except for agreement on exactly how the tests should be regarding msdos/gpt. Can someone please clarify that or I can just drop that change and we can discuss in a different PR as I would like to actually get this merged and into Fedora so it can unblock other things.

dwlehman commented:
Here is what's needed to update the tests in accordance with the rest of the PR:

```diff
diff --git a/tests/formats_test/disklabel_test.py b/tests/formats_test/disklabel_test.py
index 21c2a748..4060e731 100644
--- a/tests/formats_test/disklabel_test.py
+++ b/tests/formats_test/disklabel_test.py
@@ -70,6 +70,7 @@ class DiskLabelTestCase(unittest.TestCase):
 
         arch.is_s390.return_value = False
         arch.is_efi.return_value = False
+        arch.is_arm.return_value = False
         arch.is_aarch64.return_value = False
         arch.is_pmac.return_value = False
 
@@ -80,9 +81,16 @@ class DiskLabelTestCase(unittest.TestCase):
         arch.is_pmac.return_value = False
 
         arch.is_efi.return_value = True
+        self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt"])
+        arch.is_arm.return_value = True
         self.assertEqual(disklabel_class.get_platform_label_types(), ["gpt", "msdos"])
+        arch.is_arm.return_value = False
         arch.is_efi.return_value = False
 
+        arch.is_aarch64.return_value = True
+        self.assertEqual(disklabel_class.get_platform_label_types(), ["msdos", "gpt"])
+        arch.is_aarch64.return_value = False
+
         arch.is_s390.return_value = True
         self.assertEqual(disklabel_class.get_platform_label_types(), ["msdos", "dasd"])
         arch.is_s390.return_value = False
```

nullr0ute commented:
I split the tests off into a commit on it's own and outlined the details in the commit message, I believe this is now correct, it was also adjusted slightly to @dwlehman 's snippet above to check the right orders for the 3 different supported ARM options.

nullr0ute commented:
Jenkins, test this please.

dwlehman commented:
Jenkins, test this please.

nullr0ute commented:
@dwlehman anything more needed here? Or can it be merged?

dwlehman commented:
```
======================================================================
FAIL: test_weight_1 (devices_test.partition_test.PartitionDeviceTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/mock.py", line 1179, in patched
    return func(*args, **keywargs)
  File "/home/jenkins/workspace/blivet-3.x-PR/tests/devices_test/partition_test.py", line 318, in test_weight_1
    self.assertEqual(dev.weight, 5000)
AssertionError: 0 != 5000

======================================================================
FAIL: test_weight_2 (devices_test.partition_test.PartitionDeviceTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/blivet-3.x-PR/tests/devices_test/partition_test.py", line 373, in test_weight_2
    self.assertEqual(part.weight, spec.weight)
AssertionError: 0 != 5000

======================================================================
FAIL: test_best_label_type (formats_test.disklabel_test.DiskLabelTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/mock.py", line 1179, in patched
    return func(*args, **keywargs)
  File "/home/jenkins/workspace/blivet-3.x-PR/tests/formats_test/disklabel_test.py", line 161, in test_best_label_type
    self.assertEqual(dl._get_best_label_type(), "gpt")
AssertionError: 'msdos' != 'gpt'
- msdos
+ gpt


======================================================================
FAIL: test_platform_label_types (formats_test.disklabel_test.DiskLabelTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/mock.py", line 1179, in patched
    return func(*args, **keywargs)
  File "/home/jenkins/workspace/blivet-3.x-PR/tests/formats_test/disklabel_test.py", line 94, in test_platform_label_types
    self.assertEqual(disklabel_class.get_platform_label_types(), ["msdos"])
AssertionError: Lists differ: ['msdos', 'gpt'] != ['msdos']

First list contains 1 additional elements.
First extra element 1:
'gpt'

- ['msdos', 'gpt']
+ ['msdos']

```

nullr0ute commented:
So the last one should now be fixed, but we didn't touch the other 3 so I'm going to have to look further, I think it comes down to it's expecting what x86 wants and that's not always the case so other than shoving is_x86 around them I'm not sure how best to handle this. Suggestions welcome

nullr0ute commented:
I think these should all pass now, @dwlehman can you kick off a CI run to test please?

vojtechtrefny commented:
Jenkins, ok to test.

dwlehman commented:
```
*** Running pylint ***
PYTHONPATH=.:tests/: python3 tests/pylint/runpylint.py
************* Module /home/jenkins/workspace/blivet-3.x-PR/tests/devices_test/partition_test.py
E0001(syntax-error): 31,0: : invalid syntax (<string>, line 31)
************* Unused False Positives Found:
Instance of '(Action.*Device|Action.*Format|Action.*Member|Device|DeviceAction|DeviceFormat|Event|ObjectID|PartitionDevice|StorageDevice|BTRFS.*Device|LoopDevice)' has no 'id' member$
make[1]: *** [Makefile:61: pylint] Error 2
make[1]: Leaving directory '/home/jenkins/workspace/blivet-3.x-PR'
make[1]: Entering directory '/home/jenkins/workspace/blivet-3.x-PR'
*** Running pep8 compliance check ***
python3-pep8 --ignore=E501,E402,E731 blivet/ tests/ examples/
tests/devices_test/partition_test.py:31:1: E128 continuation line under-indented for visual indent
tests/devices_test/partition_test.py:34:1: E128 continuation line under-indented for visual indent
tests/devices_test/partition_test.py:36:5: E128 continuation line under-indented for visual indent
tests/devices_test/partition_test.py:37:9: E128 continuation line under-indented for visual indent
tests/devices_test/partition_test.py:71:17: E127 continuation line over-indented for visual indent
tests/devices_test/partition_test.py:78:17: E127 continuation line over-indented for visual indent
tests/devices_test/partition_test.py:85:17: E127 continuation line over-indented for visual indent
tests/devices_test/partition_test.py:110:5: E128 continuation line under-indented for visual indent
tests/devices_test/partition_test.py:111:9: E128 continuation line under-indented for visual indent
tests/devices_test/partition_test.py:180:5: E128 continuation line under-indented for visual indent
tests/devices_test/partition_test.py:364:1: E901 TokenError: EOF in multi-line statement
make[1]: *** [Makefile:65: pep8] Error 1
```

nullr0ute commented:
Clearly I haven't had enough coffee today :-/

nullr0ute commented:
@vojtechtrefny @dwlehman I think this should now be good :)

dwlehman commented:
Jenkins, ok to test.

nullr0ute commented:
@dwlehman @vojtechtrefny  did that CI pass? Would be really good to get this and the matching anaconda change into Fedora 29+ so it can get wider testing
